src/main/scala/com/github/anicolaspp/spark/sql/MapRDB.scala

```diff
@@ -15,15 +22,59 @@ object MapRDB {
         .schema(schema)
         .load(path)
     }
+  }
+
+
+  implicit class DataFrameOps(dataFrame: DataFrame) extends Logging {
+
+    @DeveloperApi
+    def joinWithMapRDBTable(maprdbTable: String, schema: StructType, left: String, right: String)(session: SparkSession): DataFrame = {
+
+      import org.apache.spark.sql.functions._
+      import org.ojai.store._
+
+      import collection.JavaConversions._
+      import scala.collection.JavaConverters._
+
+      val queryToRight = dataFrame
+        .cache()
+        .select(left)
+        .distinct()
 
-//    def loadFromMapRDB(path: String, schema: StructType, idxs: String*): DataFrame = {
-//      sparkSession
-//        .read
-//        .format("com.github.anicolaspp.spark.sql.Reader")
-//        .option("idx", idxs.mkString(","))
-//        .schema(schema)
-//        .load(path)
-//    }
+      val documents = queryToRight
+        .rdd
+        .mapPartitions { partition =>
+
+          if (partition.isEmpty) {
+            List.empty.iterator
+          } else {
+
+            val connection = DriverManager.getConnection("ojai:mapr:")
+            val store = connection.getStore(maprdbTable)
+
+            val m = partition.map(row =>com.mapr.db.spark.sql.utils.MapRSqlUtils.convertToDataType(row.get(0), schema.fields(schema.fieldIndex(right)).dataType) )
+              .grouped(10)
```

Comment:
I think more testing needs to be done for this to see if there is a sweet-spot for the group size and if that size changes based on the size of the join keys or some other characteristics of the right hand table.
If it looks like it is dependent on the source data then it should be exposed to the user. An unobtrusive way would be to read this from `SparkConf`